pkg/rbac/sharedData.go

```diff
@@ -140,39 +141,46 @@ func (cache *Cache) GetPropertyTypes(ctx context.Context, refresh bool) (map[str
 	}
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (cache *Cache) PopulateSharedCache(ctx context.Context) {
+	defer metric.SlowLog("PopulateSharedCache", 200*time.Millisecond)()
 	if cache.shared.isValid() { // if all cache is valid we use cache data
 		klog.V(5).Info("Using shared data from cache.")
-		return nil
-	} else { // get data and cache
+		return
+	} else { // get data and add to cache
+		var wg sync.WaitGroup
+
+		// get hub cluster-scoped resources and cache in shared.csResources
+		wg.Add(1)
+		go func() {
```

Comment:
These go-routines could have a timeout ? we may be spinning off go-routines, and they will be making many Kube-api calls , even after the client request is dead.

Reply:
Yes, the correct way to handle timeouts is with the `context` object.
The kube and postgres clients both accept a context. We just need to pass it down from the request. We have been steadily improving oour use of context, but we aren't perfect yet.